### PR TITLE
FEAT : postSlice with all CRUD ops

### PR DESCRIPTION
**Pull request by t007rushi**
## Added postSlice with thunkAPI calls for performing CRUD operations:
- create post
- delete a post
- update post by clicking edit option to update particular post by that user

### CODE REVIEW

**Review by paarthbajaj — APPROVED** [review]
Code is well structured and no suggestions found. Great efforts. You can merge.

File: src/components/CreatePost.jsx [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
+      dispatch(
+        createPost({
+          description: postData.description,
+          title: postData.title,
+          user: user.firstName,
+          userId: user.userId,
+        })
+      );
+    } catch (err) {
+      console.log(err);
```
Comment (narharikale) [inline src/components/CreatePost.jsx:24]: console.log here

File: src/pages/Profile.jsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            </svg>
+          </button>
+
+          <div
+            id="dropdown"
+            ref={optionref}
+            className={!options ?"absolute hidden z-10 w-44 text-base list-none bg-white rounded divide-y divide-gray-100 shadow dark:bg-gray-700":"absolute z-10 w-44 text-base list-none bg-white rounded divide-y divide-gray-100 shadow dark:bg-gray-700"}
+            data-popper-reference-hidden=""
+            data-popper-escaped=""
+            data-popper-placement="top"
```
Comment (MiheerTamkhane) [inline src/pages/Profile.jsx:36]: is there any need for this attributes?

File: src/pages/Profile.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                  Delete
+                </a>
+              </li>
+            </ul>
+          </div>
+        </div>
+        <div className="flex flex-col items-center pb-10">
+          <img
+            className="mb-3 w-24 h-24 rounded-full shadow-lg"
+            src=""
```
Comment (MiheerTamkhane) [inline src/pages/Profile.jsx:69]: src ?

**Review by MiheerTamkhane — APPROVED** [review]
Everything is looking good to me 
Good to merge

**Review by narharikale — APPROVED** [review]
great work rushi 💯
good to go 🚀

File: src/pages/Profile.jsx (hunk 3 above)
Comment (t007rushi) [reply src/pages/Profile.jsx:69]: Thank you for pointing out will put some default value there

File: src/pages/Profile.jsx (hunk 2 above)
Comment (t007rushi) [reply src/pages/Profile.jsx:36]: yes for interactivity

**Pull request merged**